gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -286,7 +310,16 @@ void handleQueryServiceResponseSingleColumn(
               resultKeyToAttributeMetadataMap,
               metadata,
               queryServiceValue);
-    } else { // Simple columnId Expression value eg. groupBy columns or column selections
+    } else if (mapAttributeExpression != null) {
```

Comment:
if function != null and mapAttribute != null this wouldn't be hit, right?

Reply:
For a single column, both of them can't be non-null. The function handles response for a single column.

Reply:
we can't call a function on a map attribute? e.g. `distinct_count(tags.span.type)`